### PR TITLE
Calico CI

### PR DESCRIPTION
**Pull request by Mixaster995**
part of https://github.com/networkservicemesh/integration-k8s-kind/issues/325

### CODE REVIEW

**Comment by denis-tingaikin** [comment]
@edwarnicke Finally Calico passes CI. 

As I can see this PR adds a separate job that runs a few of our tests on packet setup with Calico CNI plugin.

For me, these CI changes look overcomplicated, but I think we can go with them for now.

Can we merge this and consider simplification a bit after?

Please share your thoughts.

File: .cloudtest.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 version: 1.0
 root: "./.tests/cloud_test/"
-timeout: 7200  # 2 hour total total timeout
+timeout: 10800  # 3 hour total total timeout
```
Comment (edwarnicke) [inline .cloudtest.yaml:4]: 3 hours is a *long* timeout... why is it taking that long?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -66,8 +66,12 @@ jobs:
           git diff --name-only --exit-code go.sum || ( echo "Run go tidy" && false )
 
   packet:
-    name: packet
+    name: packet (Calico ${{ matrix.calico }})
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        calico: ["off", "on"]
```
Comment (edwarnicke) [inline .github/workflows/ci.yaml:74]: Might it not make more sense semantically to have this matrix be a matrix for CNI plugin, where calico-vpp is just one of the choices?  That would let us have an easy visible path for other CNIs to be added later.  Thoughts?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -79,6 +83,7 @@ jobs:
       - name: Install cloudtest         # 3. Install cloudtest
         run: |
           go get github.com/networkservicemesh/cloudtest@master
+#          GOPROXY=direct go get github.com/Mixaster995/cloudtest@no-cleanup
```
Comment (edwarnicke) [inline .github/workflows/ci.yaml:86]: Should this commented line be removed?

File: main_test.go [hunk 4]
```diff
@@ -17,28 +17,53 @@
...
+	if isCalico() {
+		t.Skip("not available with Calico")
+	}
 	suite.Run(t, new(multiforwarder.Suite))
 }
 
 func TestHeal(t *testing.T) {
+	if isCalico() {
+		t.Skip("not available with Calico")
+	}
```
Comment (edwarnicke) [inline main_test.go:60]: Why are heal tests not available with Calico?

File: main_test.go [hunk 5]
```diff
@@ -17,28 +17,53 @@
...
 	suite.Run(t, new(heal.Suite))
 }
+
+func TestCalico(t *testing.T) {
+	if !isCalico() {
+		t.Skip("not available without Calico")
+	}
+	suite.Run(t, new(calico.Suite))
```
Comment (edwarnicke) [inline main_test.go:68]: Why does calico have its own independent suite instead of simply running the standard test suite?

**Comment by edwarnicke** [comment]
My overall thoughts here are:

1.  It might be much more productive to think of this in terms of testing across different CNIs, rather than a binary Calico/NotCalico binary
2. I'm a little confused why Calico has its own custom suite instead of simply using the standard test suites.  I'm also confused why healing/sriov/etc are disabled.

**Pull request closed without merging**